src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc

```diff
@@ -0,0 +1,514 @@
+
+  const mobile_apis::Result::eType mobile_response_code =
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED;
+  const bool is_success = kIsNotSuccess;
+
+  ApplicationSharedPtr mock_app_empty;
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app_empty));
+
+  MessageSharedPtr msg_mobile_response =
+      CatchMobileCommandResult(CallRun(*command));
+  VerifyCommandResults(msg_mobile_response, mobile_response_code, is_success);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest, Run_NaviApplication_REJECTED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  const mobile_apis::Result::eType mobile_response_code =
+      mobile_apis::Result::REJECTED;
+  const bool is_success = kIsNotSuccess;
+
+  NaviAppSetup();
+
+  MessageSharedPtr msg_mobile_response =
+      CatchMobileCommandResult(CallRun(*command));
+  VerifyCommandResults(msg_mobile_response, mobile_response_code, is_success);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_NotMediaApplication_REJECTED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  const mobile_apis::Result::eType mobile_response_code =
+      mobile_apis::Result::REJECTED;
+  const bool is_success = kIsNotSuccess;
+
+  IncorrectAppSetup();
+
+  MessageSharedPtr msg_mobile_response =
+      CatchMobileCommandResult(CallRun(*command));
+  VerifyCommandResults(msg_mobile_response, mobile_response_code, is_success);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_VoiceCommunicationApplication_REJECTED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  const mobile_apis::Result::eType mobile_response_code =
+      mobile_apis::Result::REJECTED;
+  const bool is_success = kIsNotSuccess;
+
+  VoiceCommunicationAppSetup();
+
+  MessageSharedPtr msg_mobile_response =
+      CatchMobileCommandResult(CallRun(*command));
+  VerifyCommandResults(msg_mobile_response, mobile_response_code, is_success);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_AddIndicatorWaitingForResponseFailed_IGNORED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+  mobile_apis::AudioStreamingIndicator::eType indicator =
+      GetIndicator(msg_mobile);
+
+  const mobile_apis::Result::eType mobile_response_code =
+      mobile_apis::Result::IGNORED;
+  const bool is_success = kIsNotSuccess;
+
+  MediaAppSetup();
+
+  ON_CALL(*mock_app_, AddIndicatorWaitForResponse(indicator))
```

Comment:
@tkireva, this call looks important. Please check everywhere for this kind of issue.